### PR TITLE
meson: skip some unneeded macos-specific checks

### PR DESCRIPTION
**Pull request by Dudemanguy**
A couple of programs were always unconditionally searched for, but we don't have to do this if we're not on darwin.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -1496,6 +1487,21 @@ sys.stdout.write(swift_ver)
...
+    macos_sdk_version.version_compare('>=10.12.2'),
+    error_message: 'a suitable macos sdk version could not be found!',
+)
+
+macos_10_14_features = get_option('macos-10-14-features').require(
+    macos_sdk_version.version_compare('>=10.14'),
+    error_message: 'a suitable macos sdk version could not be found!',
+)
+
 swift = get_option('swift-build').require(
```
Comment (sfan5) [inline meson.build:1505]: you now have `swift` and `swift_build`

maybe make the checks depend on macos_sdk_version being set instead?

Comment (Dudemanguy) [reply meson.build:1505]: Great, I'm bad with names. I definitely want to essentially pass `get_option('swift-build').require('darwin')` to those checks which is different from the `swift.allowed()` check.  Not sure what to call it. `swift_opt`?

Comment (sfan5) [reply meson.build:1505]: Simpler idea, would this work?

```py
macos_sdk_version_py = find_program(join_paths(source_root, 'TOOLS',
                                    'macos-sdk-version.py'), required: darwin)
# [...]
xcrun = find_program('xcrun', required: get_option('swift-build') and darwin)
# leave the rest as-is
```

Comment (Dudemanguy) [reply meson.build:1505]: There's a slightly crappy error message if you try to enable `swift-build` on not macOS, but why would anyone do that anyways. Yeah this works.

**Comment by Dudemanguy** [comment]
>It still does this:

Flipping the check to `features['cocoa'] and cc.compiles(files(join_paths(fragments, 'touchbar.m')), name: 'macos-touchbar check')` solves that. I gave `macos_sdk_version_py` the `required: get_option('swift-build').require(darwin)` argument too so it is also skipped now.

**Pull request merged**